### ISSUE TITLE
Add vertical line to gantt plot at specified date

### ISSUE THREAD

**Issue by wcheek** [mermaid-js/mermaid#3250]
I find that the gantt plot axis is not fine grained enough, particularly when it comes to delineating the start and end of an event. I can't see the specific date when events start and end, I can only guess.

I suppose I have two feature requests regarding this issue:

1) Add a date marker at the specified date. Maybe it can be marked special somehow from the regular ticked dates, like with a special tick.
2) Add a vertical line at the specified date. Along with feature 1), this would allow a nice way to mark and specify dates of starts and ends of events.

I'm thinking something like this:

![g74](https://user-images.githubusercontent.com/66931618/180340904-849b4ab0-09b3-4e7e-a90f-250c924ed9da.png)

Generated with something like this:

```
gantt
    title Leave Plan
    dateFormat  YYYY-MM-DD
    axisFormat %b-%d
    section Time off
    Sabbatical start -- 2022-10-03       :milestone, birth, 2022-10-03, 0d
    Time off -- 2 weeks    :first_time_off, 2022-10-03, 2w
    Time off -- 2 weeks    :second_time_off, after work1, 2w
    Time off -- 3 months    :3_months_off1, after work2, 90d
    Time off -- 3 months    :3_months_off2, after work3, 90d

    section Working
    Working -- 2 weeks     :work1, after first_time_off, 2w
    Working -- 2 weeks 		:work2, after second_time_off, 2w
    Working -- 3 months     :work3, after 3_months_off1, 90d

                                       :vert, first_line, after work2
    Nov. 28                      :specify_date, nov_28_date, 2022-11-28
    
                                      :vert, second_line, after work3
    Feb. 26                       :specify_date, feb_26_date, 2023-02-26
```

I can imagine the need for new keywords, similar to milestone. Here I use `vert` and `specify_date`.

**Comment by matbun** [comment]
+1  
I was looking for the same feature!

**Comment by aliwinter** [comment]
+1

**Comment by aborruso** [comment]
+1!!

**Comment by palashkulsh** [comment]
+1

**Comment by gaardhus** [comment]
Also really agree that this would be useful

**Comment by p0lygun** [comment]
+1

**Comment by ysaereve** [comment]
+1

**Comment by StefanBrand** [comment]
You can sort-of achieve it with CSS:

```mermaid
%%{init: { 'themeCSS':
'rect[id^=vert] { height: calc(100% - 50px); transform: translate(9px, 25px); y: 0; width: 1.5px; stroke: none; fill: red; } text[id^=vert] { fill: red; y: 100%; transform: translate(-50px, 50px); font-size: 15px;}'
} }%%

gantt
    title Leave Plan
    dateFormat  YYYY-MM-DD
    axisFormat %b-%d
    section Time off
    Sabbatical start -- 2022-10-03       :milestone, birth, 2022-10-03, 0d
    Time off -- 2 weeks    :first_time_off, 2022-10-03, 2w
    Time off -- 2 weeks    :second_time_off, after work1, 2w
    Time off -- 3 months    :3_months_off1, after work2, 90d
    Time off -- 3 months    :3_months_off2, after work3, 90d

    section Working
        Working -- 2 weeks     :work1, after first_time_off, 2w
        Working -- 2 weeks 		:work2, after second_time_off, 2w
        Working -- 3 months     :work3, after 3_months_off1, 90d
        Nov. 28                            :milestone, vert1, after work2,
        Feb. 26                            :milestone, vert2, after work3,
```

**Comment by megantriplett** [comment]
Is this ticket fulfilled by the milestone https://mermaid.js.org/syntax/gantt.html#milestones or is there still a need/want for vertical lines at specified dates?
@udvale, @monicanguyen25 and I would like to work on this issue, but we want to make sure it isn't redundant first!

**Comment by StefanBrand** [comment]
First off, thank you for offering to work on this issue! 👍 

My understanding is that milestones cover the semantics of this feature, but the default styling of milestones is  a lot different. My workaround posted in https://github.com/mermaid-js/mermaid/issues/3250#issuecomment-2650460040 is not ideal because the label is dependent on the position of the milestone in the chart and might overlap with the axis labels.

In conclusion I think that this issue is mostly about an alternative styling for milestones with vertical lines and sane default positions for the labels.

**Comment by StefanBrand** [comment]
Cool, I checked it out in the netlify preview and it looks good. 👍  I prefer a different colour, but it can be easily achieved with 

```
%%{init: { 'themeCSS':
'rect.vert { stroke: red; } text.vertText { fill: red; }'
} }%%
```

**Comment by udvale** [comment]
@StefanBrand Thanks for checking it out! What color would you prefer for the vert line? Red is already used for today's marker so we thought it might clash with the line. I know on a darker theme, navy color is difficult to see. 

![Image](https://github.com/user-attachments/assets/cb02022c-304d-43de-8de4-e4f3f8dbf6ef)

**Comment by udvale** [comment]
Hey @StefanBrand, I’m only seeing the red as today’s marker, so we went ahead and implemented a different solution: the line is now navy in light theme and light blue in dark theme in which basically two different blues depending on the theme.
Let me know if that works for you!

**Comment by udvale** [comment]
Hey @StefanBrand, I’m only seeing the red as today’s marker, so we went ahead and implemented a different solution: the line is now navy in light theme and light blue in dark theme in which basically two different blues depending on the theme.
Let me know if that works for you!

<div style="display: flex; gap: 10px; align-items: center;">
  <img src="https://github.com/user-attachments/assets/9d185354-751b-4397-b6e0-bf18ca5c421b" alt="Image 1" width="250"/>
  <img src="https://github.com/user-attachments/assets/e4dd914b-b788-4ad1-9a7a-193e07313e1a" alt="Image 2" width="250"/>
</div>

**Comment by StefanBrand** [comment]
Yep, looks good. 👍
